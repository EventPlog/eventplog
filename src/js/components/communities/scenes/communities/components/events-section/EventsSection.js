import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'
import { genEventLink, genCommunityLink } from 'js/utils'

export const generateTitle = (event = {}, community = {}) => (
  <Link to={genEventLink(event, community)}>
    {event.title}
  </Link>
)

export const generateDescription = (community) => (
  <span>
    By <Link to={genCommunityLink(community)}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} people interested`
)


const EventsSection = ({ events }) => {
  const {loading, error, data } = events
  return (
    <Sidebar title="Events you may like">
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event, community);
          const description = community ? generateDescription(community) : '';
          const meta = generateMeta(event)
          return <Sidebar.Card
                    key={event.id}
                    {...{title, description, featured_image, meta}} />
        }
      )}
    </Sidebar>
  )
}

export default EventsSection