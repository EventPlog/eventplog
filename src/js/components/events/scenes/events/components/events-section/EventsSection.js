import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

// internal
import ContentPanel from 'js/components/shared/content-panel'
import Loading from 'js/components/shared/loading'
import Error from 'js/components/shared/loading/Error'
import Button from 'js/components/shared/button'
import Pagination from 'js/components/shared/pagination'
import { pluralize } from 'js/utils'

export const generateTitle = (event = {}) => {
  const community = event.community || {}
  return (
    <Link to={`/communities/${event.community_id}/events/${event.id}`}>
      {event.title}
    </Link>
  )
}

export const generateDescription = (community = {}) => (
  <span>
    By <Link to={`/communities/${community.id}`}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event = {}) => ([
  <ul key={`date${event.id}`}>
    <li>
      {event.date}
    </li>
    <li>
      {event.time}
    </li>
    <li>
      {event.venue}
    </li>
  </ul>,
  <ul key={`interest${event.id}`}>
    <li>
      {event.interested_persons} {pluralize('person', event.interested_persons)} interested
    </li>
  </ul>
])

export const generateCTA = (handleClick) => (
  <Button onClick={handleClick}>
    Interested
  </Button>
)

const EventsSection = ({
  title,
  events = {},
  attendEvent,
  getEvents,
}) => {
  const {loading, error, data = [], meta = {}} = events;
  const shouldDisplayData = (!loading && !error && data);
  return (
    <ContentPanel title={title}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {shouldDisplayData && data.map(({featured_image, ...event}) => {
          const title = generateTitle(event)
          const description = generateDescription(event.community)
          const meta = generateMeta(event)
          const btn = {onClick: () => attendEvent(event), text: 'interested'}
          const titleLink = `/communities/${event.community_id}/events/${event.id}`

          return (
            <ContentPanel.Card
              key={event.id}
              {...{title, description, featured_image, meta, btn, titleLink}}
              showButton={!event.is_attending} />
          )
        }
      )}
      {shouldDisplayData && data.length < 1 && <p>No events to display today ...</p>}
      {
        meta && meta.total_pages && (data.length > 0 || meta.current_page > 1)
          ? <Pagination totalPages={meta.total_pages}
                        activePage={meta.current_page}
                        onPageChange={getEvents} />
          : ''
      }
    </ContentPanel>
  )
}

export default EventsSection