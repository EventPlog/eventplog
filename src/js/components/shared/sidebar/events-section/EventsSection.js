import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'

export const generateTitle = (event = {}) => (
  <Link to={`/communities/${event.community_id}/events/${event.id}`}>
    {event.title}
  </Link>
)

export const generateDescription = (community) => (
  <span>
    By <Link to={`/communities/${community.id}`}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} people interested`
)


const EventsSection = ({
  title,
  events = {data: [], meta: {}},
  attendEvent
}) => {
  const {loading, error, data = {} } = events
  return (
    <Sidebar title={title || "Events you may like"}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event);
          const description = generateDescription(community);
          const meta = generateMeta(event)
          const btn = event.is_attending
                      ? {}
                      : {onClick: () => attendEvent(event), text: 'Interested'}
          return (
            <Link to={`/communities/${event.community_id}/events/${event.id}`}>
              <Sidebar.Card key={event.id}
                             {...{title, description,
                             featured_image, meta, btn}} />
            </Link>
          )
        }
      )}
    </Sidebar>
  )
}

export default EventsSection