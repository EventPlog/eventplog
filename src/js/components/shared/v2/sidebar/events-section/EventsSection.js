import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'
import { pluralize, genCommunityLink, genEventLink } from 'js/utils'

export const generateTitle = (event = {}, community = {}) => (
  <Link to={genEventLink(event, community)}>
    {event.title}
  </Link>
)

export const generateDescription = (community = {}) => (
  <span>
    By <Link to={genCommunityLink(community)}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} ${pluralize('person', event.interested_persons)} interested`
)


const EventsSection = ({
  title,
  events = {data: [], meta: {}},
  attendEvent
}) => {
  const {loading, error, data = [] } = events
  return (
    <Sidebar title={title || "Events you may like"}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event, community);
          const description = community ? generateDescription(community) : '';
          const meta = generateMeta(event)
          const titleLink = genEventLink(event, community)
          const btn = event.is_attending
                      ? {}
                      : {onClick: () => attendEvent(event), text: 'Interested'}
          return (
            <Sidebar.Card key={event.id}
                           {...{title, description, titleLink,
                           featured_image, meta, btn}} />
          )
        }
      )}
    </Sidebar>
  )
}

export default EventsSection