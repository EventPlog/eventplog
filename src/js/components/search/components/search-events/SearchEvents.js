import React from 'react'

//======== Internal Components =========
import EventsSection from 'js/components/events/scenes/events/components/events-section'
import { media } from 'js/styles/mixins'

const UserEvents = ({
  events = {},
  getEvents,
  attendEvent,
}) => {
  return (
    <div>
      <EventsSection key="upcoming-events-section"
                     title="Results" {...{events, getEvents, attendEvent }} />

    </div>
  )
}

export default UserEvents