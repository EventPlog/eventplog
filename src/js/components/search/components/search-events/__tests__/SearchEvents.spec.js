import React from 'react';
import SearchEvents from '../SearchEvents';
import EventsSection from 'js/components/events/scenes/events/components/events-section'
import { shallow } from 'enzyme';

describe('SearchEvents', () => {
  it('should render correctly', () => {
    const wrapper = shallow( <SearchEvents /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(EventsSection).length).toEqual(1)
  })

  it('should pass the right props to EventsSection', () => {
    const wrapper = shallow( <SearchEvents /> );

    // retrieve child props instances
    const eventSectionInstances = wrapper.find(EventsSection)

    // assert titles
    expect(eventSectionInstances.at(0).props().title).toEqual("Results")
  })
});