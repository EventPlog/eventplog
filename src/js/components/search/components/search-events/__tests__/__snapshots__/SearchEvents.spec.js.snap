// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchEvents should render correctly 1`] = `
<div>
  <EventsSection
    events={Object {}}
    noRecordsMsg="We couldn't find any event matching undefined"
    title="Results (0)"
  />
</div>
`;
